lightning/src/ln/channelmanager.rs

```diff
@@ -3138,6 +3165,93 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		Ok(())
 	}
 
+	/// Attempts to forward an intercepted HTLC over the provided channel id and with the provided
+	/// amount to forward. Should only be called in response to an [`HTLCIntercepted`] event.
+	///
+	/// Intercepted HTLCs can be useful for Lightning Service Providers (LSPs) to open a just-in-time
+	/// channel to a receiving node if the node lacks sufficient inbound liquidity.
+	///
+	/// To make use of intercepted HTLCs, set [`UserConfig::accept_intercept_htlcs`] and use
+	/// [`ChannelManager::get_intercept_scid`] to generate short channel id(s) to put in the
+	/// receiver's invoice route hints. These route hints will signal to LDK to generate an
+	/// [`HTLCIntercepted`] event when it receives the forwarded HTLC, and this method or
+	/// [`ChannelManager::fail_intercepted_htlc`] MUST be called in response to the event.
+	///
+	/// Note that LDK does not enforce fee requirements in `amt_to_forward_msat`.
+	///
+	/// Errors if the event was not handled in time, in which case the HTLC was automatically failed
```

Comment:
Could say "not handled in time (`HTLC_FAIL_BACK_BUFFER`)"

Reply:
`HTLC_FAIL_BACK_BUFFER` isn't `pub`, though

Reply:
Yeah neither `do_chain_event()`. Okay I find nice to give hints to the user where to look for in case of issues.